MapleServer2/Packets/GuildPacket.cs

```diff
@@ -0,0 +1,328 @@
+            pWriter.WriteByte(0x0);
+            pWriter.WriteUnicodeString(guildName);
+            return pWriter;
+        }
+
+        public static Packet UpdateGuild(GameSession session, string guildName)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.GUILD);
+            pWriter.WriteByte(0x0);
+            pWriter.WriteLong(1); // guild ID
+            pWriter.WriteUnicodeString(guildName);
+            pWriter.WriteUnicodeString(""); // guildMark Url
+            pWriter.WriteByte(0x3C); // guildC.apacity
+            pWriter.WriteUnicodeString("");
+            pWriter.WriteUnicodeString(""); // guild.Notice
+            pWriter.WriteLong(session.Player.AccountId); // guild.Leader.AccountId
+            pWriter.WriteLong(session.Player.CharacterId); // guild.Leader.CharacterId
+            pWriter.WriteUnicodeString(session.Player.Name); // guild.Leader.Player.Name
+            pWriter.WriteLong(1609108139); // guild creation timestamp
+            pWriter.WriteByte(0x0);
+            pWriter.WriteInt(1000);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0);
+            pWriter.WriteByte(0x1);
+            pWriter.WriteInt(0);
+            pWriter.WriteInt(0); // guild.Exp
+            pWriter.WriteInt(0); // guild.Funds
+            pWriter.WriteByte(0x0);
+            pWriter.WriteInt(0);
+            pWriter.WriteByte(0x0); // guild.Members.Count and also loop for each member ?
+            pWriter.WriteByte(0x6); // loop for guild ranks ?
+            pWriter.WriteByte(0x0);
+            pWriter.WriteUnicodeString("Leader");
+            pWriter.WriteInt(4095);
+            pWriter.WriteByte(0x1);
+            pWriter.WriteUnicodeString("Jr.Leader");
+            pWriter.WriteInt(1);
+            pWriter.WriteByte(0x2);
+            pWriter.WriteUnicodeString("Member");
+            pWriter.WriteInt(1);
+            pWriter.WriteByte(0x3);
+            pWriter.WriteUnicodeString("NewMember");
+            pWriter.WriteInt(1);
+            pWriter.WriteByte(0x4);
+            pWriter.WriteUnicodeString("Member2");
+            pWriter.WriteInt(1);
+            pWriter.WriteByte(0x5);
+            pWriter.WriteUnicodeString("NewMember2");
+            pWriter.WriteInt(1);
+            pWriter.WriteByte(0x9); // loop for all guild skills
+            pWriter.WriteInt(1); // guild skill ID
+            pWriter.WriteInt(1); // guild skill level
+            pWriter.WriteLong(0);
+            pWriter.WriteInt(2);
+            pWriter.WriteInt(1);
+            pWriter.WriteLong(0);
+            pWriter.WriteInt(3);
+            pWriter.WriteInt(1);
+            pWriter.WriteLong(0);
+            pWriter.WriteInt(4);
+            pWriter.WriteInt(1);
+            pWriter.WriteLong(0);
+            pWriter.WriteInt(10001);
```

Comment:
I'd make these into a loop of some sort also

Reply:
these are part of the same guild skills loop